python/kvikio/nvcomp_codec.py

```diff
@@ -226,20 +231,23 @@ def decode_batch(
             self._stream,
         )
 
+        # If all chunks are the same size, we can just return uncomp_chunks.
+        if is_equal_chunks and out is None:
+            return cp.asnumpy(uncomp_chunks) if is_host_buffer else uncomp_chunks
+
         res = []
         for i in range(num_chunks):
-            ret = uncomp_chunks[i]
-            if out is not None and out[i] is not None:
+            ret = uncomp_chunks[i, : uncomp_chunk_sizes[i].item()]
```

Comment:
<!-- Polish comments are like a suggestion, where there is nothing necessarily wrong with the relevant content, there's just some ways to immediately improve the quality.  -->
**polish:** ‏Avoid excessive d2h synchronisation by copying `uncomp_chunk_sizes` to the host in one go before this loop?

Reply:
Done.